activitysim/core/cleaning.py

```diff
@@ -32,24 +34,35 @@ def recode_to_zero_based(values, mapping):
     return result
 
 
-def should_recode_based_on_table(tablename):
+def should_recode_based_on_table(state: workflow.State, tablename):
     try:
-        base_df = inject.get_table(tablename).to_frame()
+        base_df = state.get_dataframe(tablename)
     except (KeyError, RuntimeError):
         # the basis table is missing, do not
```

Comment:
Yes, the land_use_taz content is optional unless you are recoding zone numbers, and the recoding was introduced with Sharrow, although in theory you can apply the recoding and not otherwise use sharrow (but I'm not sure why you would want to).

I've modified the land_use_taz table to automatically infer the TAZs from the land use table instead of logging an error that is then handled, although there will need to be some consistency with the skims -- the TAZs must match, and I think that consistency is currently trust-based and not otherwise checked.

Reply:
Suggest adding an assert statement when inferring TAZs from the land use table (and in general?) that the number of TAZs matches the shape of the skims.